python/ray/remote_function.py

```diff
@@ -254,6 +252,15 @@ def _remote(self, args=None, kwargs=None, **task_options):
         worker = ray._private.worker.global_worker
         worker.check_connected()
 
+        # We cannot do this when the function is first defined, because we need
+        # ray.init() to have been called when this executes
+        if not self._injected_tracing:
+            self._function = _inject_tracing_into_function(self._function)
+            self._function_signature = ray._private.signature.extract_signature(
+                self._function
+            )
+            self._injected_tracing = True
```

Comment:
the ray API needs to be thread-safe, is there any concern of a race condition here if multiple threads invoke a function concurrently?

Reply:
Actually no `_inject` is not idempotent so